src/psp.ts

```diff
@@ -0,0 +1,664 @@
+          };
+          const authorized =
+            isEmpty(chain.authorizers) || chain.authorizers.some(authorizesRequesterForAirnodeEndpoint);
+          if (!authorized) {
+            node.logger.info('Requester not authorized', subscriptionIdLogOptions);
+            continue;
+          }
+          */
+
+            // **************************************************************************
+            // Fetch template details
+            // **************************************************************************
+            node.logger.debug('Fetching template details...', subscriptionIdLogOptions);
+
+            const template = templates[templateId];
+            if (isNil(template)) {
+              node.logger.warn(`TemplateId ${templateId} not found in templates`, subscriptionIdLogOptions);
+              continue;
+            }
+
+            // **************************************************************************
+            // Make API call
+            // **************************************************************************
+            node.logger.debug('Making API request...', subscriptionIdLogOptions);
+
+            const [errorCallApi, logsData] = await retryGo(() =>
+              callApi({
+                airnodeAddress,
+                oises,
+                apiCredentials,
+                id: subscriptionId,
+                templateId,
+                overrideParameters,
+                oisTitle,
+                endpointName,
+                endpointId: template.endpointId,
+                templateParameters: template.templateParameters,
+              })
+            );
+            if (!isNil(errorCallApi) || isNil(logsData)) {
+              node.logger.warn('Failed to fecth API value', subscriptionIdLogOptions);
+              continue;
+            }
+            const [logs, data] = logsData;
+            node.logger.logPending(logs, subscriptionIdLogOptions);
+
+            const apiValue = data[subscriptionId];
+            if (isNil(apiValue)) {
+              node.logger.warn('API value not found. Skipping update...', subscriptionIdLogOptions);
+              continue;
+            }
+
+            // **************************************************************************
+            // Check conditions
+            // **************************************************************************
+            node.logger.debug('Checking conditions...', subscriptionIdLogOptions);
+
+            const encodedFulfillmentData = ethers.utils.defaultAbiCoder.encode(['int256'], [apiValue]);
+            let conditionFunction: ethers.utils.FunctionFragment;
+            let conditionParameters: string;
+            try {
+              const decodedConditions = abi.decode(conditions);
+              const [decodedConditionFunctionId] = ethers.utils.defaultAbiCoder.decode(
+                ['bytes4'],
+                decodedConditions._conditionFunctionId
+              );
```

Comment:
Note that Airnode ABI only supports `bytes32`. So I'd decode `bytes32` and get rid of the leading zeros manually to get the `bytes4`.

Reply:
Is this how I'm supposed to be encoding the `conditionFunctionId`?:

```
  const beaconUpdateSubscriptionConditions = [
    {
      type: 'bytes32',
      name: '_conditionFunctionId',
      value: hre.ethers.utils.defaultAbiCoder.encode(
        ['bytes4'],
        dapiServer.interface.getSighash('conditionPspBeaconUpdate')
      ),
    },
    { type: 'bytes', name: '_conditionParameters', value: beaconUpdateSubscriptionConditionParameters },
  ];
```
If so then I could be sure that the encoded string within the `bytes32` is a `bytes4`, right?

Reply:
Right